server/commands/teamCreator.ts

```diff
@@ -50,7 +50,7 @@ async function teamCreator({
       name,
       avatarUrl,
       authenticationProviders,
-    },
+    } as Partial<InferCreationAttributes<Team>>,
```

Comment:
Yeah, this isn't good but can't seem to find a better way to fix this. We have instances where `teamProvisioner`/`teamCreator` are called with only `{ name: string, providerId: string }` as `authenticationProvider`. One such example is [here](https://github.com/outline/outline/blob/21a1257d0652cb3d043c99b13db5884f3206708d/plugins/azure/server/auth/azure.ts#L112). Makes it hard to do type plumbing. Also, gave a try to overriding `Team.create` with modified type only to complicate it further 😞